extensions/emmet/src/matchTag.ts

```diff
@@ -4,42 +4,46 @@
  *--------------------------------------------------------------------------------------------*/
 
 import * as vscode from 'vscode';
-import { HtmlNode } from 'EmmetNode';
-import { getHtmlNode, parseDocument, validate } from './util';
-
+import { toLSTextDocument, validate, getHtmlNodeLS, offsetRangeToSelection } from './util';
+import { TextDocument as LSTextDocument } from 'vscode-html-languageservice';
 
 export function matchTag() {
 	if (!validate(false) || !vscode.window.activeTextEditor) {
 		return;
 	}
 
 	const editor = vscode.window.activeTextEditor;
-	let rootNode: HtmlNode = <HtmlNode>parseDocument(editor.document);
-	if (!rootNode) { return; }
+	const document = toLSTextDocument(editor.document);
 
 	let updatedSelections: vscode.Selection[] = [];
 	editor.selections.forEach(selection => {
-		let updatedSelection = getUpdatedSelections(editor, selection.start, rootNode);
+		const updatedSelection = getUpdatedSelections(document, selection.start);
 		if (updatedSelection) {
 			updatedSelections.push(updatedSelection);
 		}
 	});
-	if (updatedSelections.length > 0) {
+	if (updatedSelections.length) {
 		editor.selections = updatedSelections;
 		editor.revealRange(editor.selections[updatedSelections.length - 1]);
 	}
 }
 
-function getUpdatedSelections(editor: vscode.TextEditor, position: vscode.Position, rootNode: HtmlNode): vscode.Selection | undefined {
-	let currentNode = getHtmlNode(editor.document, rootNode, position, true);
-	if (!currentNode) { return; }
+function getUpdatedSelections(document: LSTextDocument, position: vscode.Position): vscode.Selection | undefined {
+	const currentNode = getHtmlNodeLS(document, position, true);
+	if (!currentNode) {
+		return;
+	}
+
+	const offset = document.offsetAt(position);
 
 	// If no closing tag or cursor is between open and close tag, then no-op
-	if (!currentNode.close || (position.isAfter(currentNode.open.end) && position.isBefore(currentNode.close.start))) {
+	if (!currentNode?.endTagStart
```

Comment:
currentNode is defined here, so the '?' can be removed.
I thory 0 is a valid offset, admittedly not possible for the tagStart. So better `currentNode.endTagStart === undefined`